contracts/interfaces/ICover.sol

```diff
@@ -9,27 +9,44 @@ interface ICover is IERC721 {
   /* ========== DATA STRUCTURES ========== */
 
   struct StakingPool {
-    uint160 id;
-    uint96 bookedAmount;
+    address poolAddress;
+    uint96 coverAmount;
+    uint96 premiumInNXM;
   }
 
   struct Cover {
     uint24 productId;
     uint8 payoutAsset;
-    uint8 deniedClaims;
     uint96 amount;
     uint32 start;
     uint32 period;  // seconds
+    uint96 premium;
   }
 
   struct Product {
```

Comment:
needs and ipfs hash as well for things like product name, any potential exclusions / inclusions from the main cover wording of the product type

Reply:
@roxdanila here we have 

```  struct ProductType {
    string descriptionIpfsHash;
    uint8 redeemMethod;
    uint16 gracePeriodInDays;
    uint16 burnRatio;
  }```
  
  where that hassh is contained.
  
  The product points to it through the productType

Reply:
True, but that has generic "product type" information.
I argue that for info specific to the product ID (i.e. specific exclusions for sushiswap for example), we'd need something in the product struct as well.